### PR TITLE
CT-180 Add subject to correspondence

### PR DESCRIPTION
**Pull request by EdwardAndress**
https://dsdmoj.atlassian.net/browse/CT-180

Add subject to correspondence
Allow input on form for new FOI
Show value in list view

### CODE REVIEW

File: spec/features/02_create/correspondence_create_spec.rb [hunk 1]
```diff
@@ -3,9 +3,10 @@
 feature 'Correspondence can be created ' do
 
   given(:correspondence) do
-    Struct.new('Correspondence', :name, :email, :message).new(
+    Struct.new('Correspondence', :name, :email, :subject, :message).new(
```
Comment (misaka) [inline spec/features/02_create/correspondence_create_spec.rb]: 👍 I like this use of `Struct`.

File: spec/models/correspondence_spec.rb [hunk 2]
```diff
@@ -60,8 +61,13 @@
     end
   end
 
-  describe 'associations' do
+  describe '#subject' do
+    it { should validate_length_of(:subject).is_at_most(80) }
+    it { should allow_value('a' * 80).for :subject          }
```
Comment (misaka) [inline spec/models/correspondence_spec.rb]: Why the doubling up of tests here?

**Comment by EdwardAndress** [comment]
@misaka - did you make a note somewhere of your 'question about the tests'?  I am going to merge this now but am happy to make changes afterwards to address whatever concern you may have.

**Comment by misaka** [comment]
@EdwardAndress It's [in the review](https://github.com/ministryofjustice/correspondence_tool_staff/pull/39/files/33fc1ec3f43b9324e8c6c0b366b88d11d1197f51#diff-317fdef0e11af3fa3f4b91221e8c9a4bR66)

**Comment by EdwardAndress** [comment]
Ah, yes, sorry I was in a rush and missed it.  The doubling was an artefact of some debugging that I had to do.  Turned out that the bug was me.  I have removed the duplicative / triplicative tests.

**Comment by misaka** [comment]
@EdwardAndress - 👍 Thanks, good to go!

**Pull request merged**